docs: document the behavior of advanced templating for custom targets (#5810)
external-secrets/external-secrets: docs/guides/targeting-custom-resources.md

The `target` field accepts dot-notation paths like `spec.database` or `spec.logging` to place the rendered template output at specific locations in the resource structure. When `target` is not specified it defaults to `Data` for backward compatibility with Secrets.

!!! note "Using `property` when templating `data`"

>     The return of `data:` find is a map. If templated as a `string` it will fail in finding the right key. Therefore, something like this:

Reviewer: ```suggestion
    The return of `data:` isn't an object on the template scope. If templated as a `string` it will fail in finding the right key. Therefore, something like this:
```
